04.bowling/bowling.rb

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+score = ARGV[0]
+scores = score.split(',')
+
+shots = []
+
+scores.each do |s|
+  if s == 'X'
+    shots << 10
+    shots << 0
+  else
+    shots << s.to_i
+  end
+end
+
+frames = shots.each_slice(2).to_a
+
+# 第10フレームは最大3投になるので、3投分の点数を平坦化して配列に格納
+# ex [10, 0, 10, 10] => [10, 10, 10]に変更する
+frame_10th_processing = frames[9..11].flatten
+frame_10th = frame_10th_processing.reject(&:zero?)
+
+frame_1st_to_9th = frames[0..8]
+all_frames = [*frame_1st_to_9th << frame_10th]
```

Comment:
`frame_1st_to_9th`は0〜8の意味を1〜9に置き換えただけなので、無理に変数に入れなくていいかもしれません。
（コード上、9が行方不明になるデメリットはありますが、プログラマなら十分理解できるはずなので許容範囲かと）

あと、`*`を使うのであれば`<<`は不要です。

```suggestion
all_frames = [*frames[0..8], frame_10th]
```

Reply:
こちらの処理で配列に`*`をつけて非破壊的処理で連結する目的でしたが、`<<`で破壊的変更をしてしまってました😓
意図していたのは、ご指摘どおりの処理でした。。